### ISSUE TITLE
Playground: language changing broken?

### ISSUE THREAD

**Issue by tomByrer** [sf-web/sf-web.github.io#8]
When I change the language, nothing changes but the highlighting.  Used to be different code examples were swapped?
Win & Linux, Chrome, Fx, IE

**Comment by chrissimpkins** [comment]
I don't believe that @jublo implemented different example code blocks for different languages, just the JS specimen and option to enter your own, then highlight it to date.

**Comment by jublo** [comment]
@tomByrer Chris is right, for now there are no additional code samples.  Do you have any suitable ones in mind?

**Comment by tomByrer** [comment]
Thanks for your interest!
I found a decent code example collection:
https://github.com/isagalaev/highlight.js/blob/master/test/detect/css/default.txt
But most are missing a few essentials like comments.  I'm thinking of splitting the examples into a separate repo & massage them there.  I haven't decided on the best file structure / naming convention.  Or maybe there is a better collection? What do you guys think?

**Comment by chrissimpkins** [comment]
@tomByrer :+1:  Like the idea.  @jublo thoughts?

**Comment by jublo** [comment]
@tomByrer Thanks for the effort you put into this!  We’ll gladly take any code you can provide, but the better prepared, the less work for us! :wink:

**Comment by chrissimpkins** [comment]
@jublo @tomByrer have a look at https://github.com/source-foundry/code-corpora 

We have a number of commonly used projects across common language types that we are using as the basis of our source code analysis to optimize Hack.  Perhaps a place to pinch some code and get started?

**Comment by tomByrer** [comment]
That looks great @chrissimpkins; exactly what I was looking for!
I discovered a few extras for commonly confused chars here:
https://github.com/source-foundry/text-specimens/blob/master/semi-source/confusedchar-compact.txt

**Comment by chrissimpkins** [comment]
@tomByrer it is pseudo code and may cause syntax highlighting issues but it is a great example of characters that are commonly confused.  thoughts @jublo?

**Comment by chrissimpkins** [comment]
I have been tossing around the idea of generating short code snippets for the most common programming languages as part of that repo that you linked Tom.  Maybe a better approach would be to understand the common tokens that are used in each language and then find a way to display those along with commonly confused characters in each specimen.  We can be creative.  I think that this would be a great resource for others and would support the syntax highlighting in the ACE editor that JM developed here.

**Comment by tomByrer** [comment]
> common tokens that are used in each language and then find a way to display those along with commonly confused characters in each specimen

That sounds like a good idea.

**Comment by chrissimpkins** [comment]
@tomByrer Here is where we are pulling together the source code specimens

https://github.com/source-foundry/code-corpora 

I've put together an early version of a script that analyzes the token frequency in each of the language types with the Python NLTK library and intend to expand on this analysis.  This was prompted by our interest in a more objective approach to the optimization of the Hack typeface and we intend to use the results to guide upcoming changes to the Hack fonts.

I think that the analysis will be very helpful for the development of a set of source code text specimen "standards" as well.  We could place these out in the wild in an attempt to reach a consensus among the open source developer community.

**Comment by jublo** [comment]
@chrissimpkins The confusedchar file looks good, especially in view of our discussions about backslashes, pipes etc.

**Comment by chrissimpkins** [comment]
@jublo question is about implementation on Font Playground as a new specimen type

**Comment by jublo** [comment]
@chrissimpkins Elaborate.

**Comment by chrissimpkins** [comment]
@jublo see OP message and your post earlier in this thread.  he suggested code for Font Playground

**Comment by jublo** [comment]
@chrissimpkins Are we to swap sample code shown on editors once user changes selected language?

**Comment by chrissimpkins** [comment]
@jublo Your font playground design.  Please implement what you feel best, PR, and we will get it merged in.  Code samples still exist in that code corpora repository if they are helpful.

**Comment by jublo** [comment]
[ ] Ask user whether they want to keep snippet edits (if any)
[ ] update code sample as per selected language

**Comment by chrissimpkins** [comment]
add `-` in front of your boxes and you will get true checklist Markdown that you can check:

- [ ] Ask user whether they want to keep snippet edits (if any)
- [ ] update code sample as per selected language

If you put them in the OP, you see the number of incomplete checklist tasks in the IR listing on Github.  Not sure if that feature works from thread posts or not (we'll see after this is entered)

**Comment by jublo** [comment]
@chrissimpkins Apparently thanks!
